### PR TITLE
Refactor gateway server tls adjugement 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

1,  IsTLSServer is very hard to understand.  Rename to IsTLSServerForNonHTTP
2. Add IsHTTPSServerWithTLSTermination to replace gateway.IsTLSServer(server) && gateway.IsHTTPServer(server)

### CODE REVIEW

File: pilot/pkg/model/gateway_test.go [hunk 1]
```diff
@@ -142,12 +142,14 @@ func TestMergeGateways(t *testing.T) {
 				t.Errorf("Incorrect number of routes. Expected: %v Got: %d", len(tt.serversForRouteNum), len(mgw.ServersByRouteName))
 			}
 			for k, v := range mgw.ServersByRouteName {
+				fmt.Println("----route  ", k, "servers ", v)
```
Comment (ramaraochavali) [inline pilot/pkg/model/gateway_test.go:145]: nit: remove

File: pilot/pkg/model/gateway_test.go [hunk 2]
```diff
@@ -142,12 +142,14 @@ func TestMergeGateways(t *testing.T) {
...
 				if tt.serversForRouteNum[k] != len(v) {
 					t.Errorf("for route %v expected %v servers got %v", k, tt.serversForRouteNum[k], len(v))
 				}
 			}
 			ns := 0
 			for _, ms := range mgw.MergedServers {
+				fmt.Println("-----", ms.Servers)
```
Comment (ramaraochavali) [inline pilot/pkg/model/gateway_test.go:152]: nit: remove

File: pkg/config/gateway/gateway.go [hunk 3]
```diff
@@ -20,14 +20,25 @@ import (
 	"istio.io/istio/pkg/config/protocol"
 )
 
-// IsTLSServer returns true if this server is non HTTP, with some TLS settings for termination/passthrough
-func IsTLSServer(server *v1alpha3.Server) bool {
+// IsTLSServerForNonHTTP returns true if this server is non HTTP, with some TLS settings for termination/passthrough
+func IsTLSServerForNonHTTP(server *v1alpha3.Server) bool {
```
Comment (ramaraochavali) [inline pkg/config/gateway/gateway.go:24]: IsNonHTTPTLSServer may sound better

Comment (hzxuzhonghu) [reply pkg/config/gateway/gateway.go:24]: SGTM

File: pilot/pkg/model/gateway_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/model/gateway_test.go]: done

File: pilot/pkg/model/gateway_test.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/model/gateway_test.go]: done

File: pkg/config/gateway/gateway_test.go [hunk 4]
```diff
@@ -39,7 +39,7 @@ func TestIsTLSServerForNonHTTP(t *testing.T) {
 				},
 				Tls: &v1alpha3.ServerTLSSettings{HttpsRedirect: true},
 			},
-			expected: true,
+			expected: false,
```
Comment (ramaraochavali) [inline pkg/config/gateway/gateway_test.go:42]: Why is this changed? Was it a bug earlier?

Comment (hzxuzhonghu) [reply pkg/config/gateway/gateway_test.go:42]: Prvious commit changed it accidentally. And this is a revert

**Review by ramaraochavali — APPROVED** [review]
